SQL or NoSQL

kimkimani opened the pull request:
### Article Checklist
#### Preliminary Checks - Formatting and Structure 
- [x] Does your article follow any of the suggested structure formats? see the example [Formats here](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)
- [x] Is your article properly formatted in Markdown? - see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)
- [x] Have you used the correct folder and file structure? - see [Contribution Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)
- [x] Is your article over 750 words?
- [x] Have you used our preferred heading structure? - [H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)
- [x] Does your article provide enough value and detail about your topic? Articles should be clear, accurate, and fully explained.
- [x] Can your article be understood by beginners? **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image? (600X400 pixels (images can be resized in paint) - also please make sure the image is from a site like www.unsplash.com)
- [x] In your PR - add a comment that states ["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278) to link and close your Idea Content (issue) when the PR is merged.
      

#### Grammar Checks
- [x] Have you spellchecked and grammar checked your article? - Tool: [Grammarly](https://grammarly.com)
- [x] Have you checked your article for readability? - Tool: [Hemmingway](http://www.hemingwayapp.com/)
- [x] Have you added sources for quotes and images that aren't yours?

### Technical Checks
- [x] Are your code snippets properly formatted for syntax highlighting - see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting/)
- [ ] Have you used Repl.it for your code snippets?

### Contribution Guidelines
For first-time contributors and for more details, see [Contributing Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)
Fixes #345

cyberShaw commented:
@hectorkambow This article needs front matter and a hero image. Please add them to this article 😄

kimkimani commented:
> @hectorkambow This article needs front matter and a hero image. Please add them to this article 😄

@cyberShaw , am designing a hero image, I will upload it once done. Thank you

cyberShaw commented:
> > @hectorkambow This article needs front matter and a hero image. Please add them to this article 😄
> 
> @cyberShaw , am designing a hero image, I will upload it once done. Thank you

That sounds great!

cyberShaw commented on articles/sql-or-nosql-when-to-choose-what/index.md:
```diff
@@ -0,0 +1,123 @@
+
+Town A represents the relational SQL, and Town B represents the non-relational NoSQL. SQL is powerful and versatile. On the other hand, they are restrictive and structured with a predefined schema. If you choose to use them, your data must use and follow the same structure. And as in Town A, changing its structure would be difficult. It will disrupt your whole system. The language of SQL is tabul ...
+
+NoSQL on the hand, Just like town B, it has a dynamic unstructured schema that is flexible to any new resident. Its data can be stored in many ways. This data can be
```
**NoSQL on the hand, Just like town B**

"NoSQL on the other hand, like town B" would sound better.

cyberShaw commented on articles/sql-or-nosql-when-to-choose-what/index.md:
```diff
@@ -0,0 +1,123 @@
+
+#### Scalability
+
+Assume and figure out a building near your neighbourhood. Be the architecture and check the possibility of adding more floors to the existing building to host more rooms. Would you propose adding more floors to the existing building or would you build a new one? If you will add more floors, what are the chances of this building to remain stable? That the case here. SQL is scaled vertically, which ...
```
**Assume and figure out a building near your neighbourhood.**

"Think of a building near your neighbourhood" would flow better.

cyberShaw replied:
**Be the architecture**

Here, "Be the architect" sounds more appropriate.

cyberShaw replied:
**That is one of the restrictions of a SQL database approach that makes scaling to be super hard.**

"That is one of the restrictions of an SQL database approach that makes scaling hard" flows better.

cyberShaw commented on articles/sql-or-nosql-when-to-choose-what/index.md:
```diff
@@ -0,0 +1,123 @@
+Let’s look at the 2020 [annual developer survey](https://insights.stackoverflow.com/survey) trends conducted by StackOverflow involving over 65,000 respondents from over 180 countries.
+
+- *[Most popular](https://insights.stackoverflow.com/survey/2020#technology-databases-all-respondents4) databases*
+
```
You could add a table that concludes by summarizing the differences between SQL and NoSQL databases.

To add a table in Markdown, check out the syntax [here](https://www.markdownguide.org/extended-syntax/#tables).

cyberShaw replied:
@kimkimani Please do add a summary of the things that were discussed in the form of a table, this would ensure that readers end up with a crisp summary in their mind.

kimkimani replied:
@cyberShaw A table seems to be very much repetitive, I consider adding a conclusion instead, will that be okay?

cyberShaw replied:
Yep, that works too.

cyberShaw commented on articles/sql-or-nosql-when-to-choose-what/index.md:
```diff
@@ -0,0 +1,123 @@
+SQL statements will help you to perform CRUD operations such as CREATE, READ, UPDATE, and DELETE. These operations will help you to retrieve and modify data.
+If we were to have a student info database with table student and course, this is how we can represent a relation in an SQL database.
+
+![image title](\tables.png)
```
You need to add a meaningful image title to every image that you've included.

Example, [SQL Tables]

cyberShaw commented on articles/sql-or-nosql-when-to-choose-what/index.md:
```diff
@@ -0,0 +1,123 @@
+
+It is not like the early times when data was stored on papers and cabinet. Data is now all around us. Information is the backbone of almost every modern business models. With the current steep growth of the database technology, as a developer building an application to store some data online, and already know what a database is, the big question you should be asking yourself is which database mod ...
+SQL or NoSQL? That is what we will dive in briefly, to understand their structures, explore them deeply, and understand how they work. These two databases do one thing, to store data, but the approach to do this may differ.
```
**That is what we will dive in briefly, to understand their structures, explore them deeply, and understand how they work.**

"Let's dive into SQL vs NoSQL, understand their structures, explore them in-depth, and understand how they work." flows better

cyberShaw replied:
**These two databases do one thing, to store data, but the approach to do this may differ.**

"SQL and NoSQL databases store data, but the approach they take differs" would flow better.

cyberShaw commented on articles/sql-or-nosql-when-to-choose-what/index.md:
```diff
@@ -0,0 +1,123 @@
+
+### What SQL
+
+SQL is the most popular database structure. Which I too know at one instance heard of it. SQL (commonly pronounced as "see quell") stands for the structured query language.  SQL is a language used for writing query statements to interact and communicate with relational databases. A relational database is a database that has a tabular schema to store data in terms of rows, and columns. Each column ...
```
**Each column contains data and each row is called a record.**

"Each column contains data that are called attributes and each row is called a record." explains the point better.

cyberShaw commented on articles/sql-or-nosql-when-to-choose-what/index.md:
```diff
@@ -0,0 +1,123 @@
+
+![image title](\tables.png)
+
+![image title](\sql.png)
```
The image path format should be given in the form,

```
![image title](/engineering-education/new-folder-name/image-name.jpg)
```

Please change the image paths for all your images. And please do check out the path and folder structure guidelines [here](https://github.com/section-io/engineering-education/blob/master/UPLOAD_INSTRUCTIONS.md)

cyberShaw commented on articles/sql-or-nosql-when-to-choose-what/index.md:
````diff
@@ -0,0 +1,123 @@
+Like SQL, NoSQL has a unique id that uniquely identifies a specific object within a collection. Nested objects and array are supported where an object can be inside another object. If we were to represent the SQL student info database in NoSQL, this how it would look like:
+
+```json
+Student {
````
Since you have displayed your data in JSON, every key and value should be enclosed in quotes. For the example you've given,

```json
{
  "Student": {
    "_id": "Objectld('14s5gg98we9d’)",
    "firstName": "Alex",
    "lastName": "Brooks",
    "username": "alex09_b",
    "Courses": "[{name: Web Dev}, {name: React js}]"
  }
}
```

cyberShaw commented on articles/sql-or-nosql-when-to-choose-what/index.md:
```diff
@@ -0,0 +1,123 @@
+
+#### Relation
+
+SQL employs a kind of relationship that exists between two or more database tables. As we have explained above, NoSQL is non-relational. To get this point correct, think of two towns let's say town A and town B. Take town A, where the residents understand only one language. They use it to do all sorts of communication and interaction around them day in day out. To change the language of this town ...
```
**To get this point correct,**

Here, "To understand this point" sounds more appropriate.

cyberShaw commented on articles/sql-or-nosql-when-to-choose-what/index.md:
```diff
@@ -0,0 +1,123 @@
+
+#### Scalability
+
+Assume and figure out a building near your neighbourhood. Be the architecture and check the possibility of adding more floors to the existing building to host more rooms. Would you propose adding more floors to the existing building or would you build a new one? If you will add more floors, what are the chances of this building to remain stable? That the case here. SQL is scaled vertically, which ...
```
**Assume and figure out a building near your neighbourhood.**

"Think of a building in your neighbourhood" flows better.

cyberShaw replied:
**Be the architecture** 

"Be the Architect" is the correct term and would be more suitable.

cyberShaw reviewed (changes requested):
Great article about databases, @kimkimani !

It was full of great content and your use of analogies for explaining the structured nature of SQL and scaling was brilliant! 

I've done a check and reviewed for readability and grammar and have suggested some changes. A few points:

- Please do a grammar check using a tool like [Grammarly](http://www.grammarly.com/)
- Split large sentences into smaller ones for better flow and readability. Use http://www.hemingwayapp.com/ for checking the same.

Please make the suggested changes and we can move on to the tech review!

kimkimani commented:
@cyberShaw  Should flow better now

cyberShaw reviewed (changes requested):
Thank you for making the changes I suggested, @kimkimani !
I've gone through the changes and added a commit to improve readability and fixed some grammatical errors.

Please make these other changes that I suggested and we can move on the tech review stage! 😄

cyberShaw commented on articles/sql-or-nosql-when-to-choose-what/index.md:
```diff
@@ -0,0 +1,123 @@
+
+It is not like the early times when data was stored on papers and cabinet. Data is now all around us. Information is the backbone of almost every modern business models. With the current steep growth of the database technology, as a developer building an application to store some data online, and already know what a database is, the big question you should be asking yourself is which database mod ...
+SQL or NoSQL? That is what we will dive in briefly, to understand their structures, explore them deeply, and understand how they work. These two databases do one thing, to store data, but the approach to do this may differ.
+
```
Please add a table of contents too @kimkimani

kimkimani commented:
@cyberShaw Added some enhancements

kimkimani commented:
@cyberShaw hopes add some improvements

kimkimani commented on articles/sql-or-nosql-when-to-choose-what/index.md:
```diff
@@ -80,34 +87,38 @@ If these admins walked into an SQL bar, they would have defiantly found a table
 Think of a building near your neighborhood. Be the architect and check the possibility of adding more floors to the existing building to host more rooms. Would you propose adding more floors to the existing building, or would you build a new one? If you will add more floors, what are the chances of this building to remain stable? That the case here.
 
-SQL is scaled vertically. This means that a server capacity can be increased by increasing specks such as CPU, RAM, and disk size (HDD or SDD) (more floors will be added to the existing building). Vertical scaling adds more power to your existing server. Of course, the downside of that is that there will be a limit point hit on how much power you can stretch your server. That is one of the restri ...
+SQL is scaled vertically. This means that a server capacity can be increased by increasing specks such as CPU, RAM, and disk size (HDD or SDD) (more floors will be added to the existing building). Vertical scaling adds more power to your existing server. Of course, the downside of that is that there will be a limit point hit on how much power you can stretch your server. That is one of the restri ...
```
@cyberShaw Added some links to the abbreviations

cyberShaw commented:
Great job on adding the requested changes! @kimkimani 

I've added a few more minor changes in commit: 61a2aed

This article can now move on to the tech review phase 😄

cyberShaw commented on articles/sql-or-nosql-when-to-choose-what/index.md:
```diff
@@ -1,36 +1,37 @@
 
 ![SQL database table](/engineering-education/sql-or-nosql-when-to-choose-what/tables.png)
 
 ![SQL](/engineering-education/sql-or-nosql-when-to-choose-what/sql.png)
```
Please add attributions to all of the image sources.

cyberShaw reviewed (changes requested):
Everything looks good @kimkimani 
For the tech review phase, I've suggested some minor changes. Please add them and we can mark this article ready for the final review.

kimkimani commented:
Thank you @cyberShaw

kimkimani commented:
@cyberShaw  made some updates

cyberShaw commented:
@kimkimani Everything looks good, let's move this article for final review!
@hectorkambow this article is ready for the final review

hectorkambow commented:
Plagiarism check done

hectorkambow commented:
@kimkimani  great article - seems very informative - learned a lot about Nosql and SQL - wanted to ask if any of the other images needed sources (for example the joke image)

Please add those and then we can push your article to tech review 🚀 

@cyberShaw - great work on the review - I added the peer review line at the bottom take a look and be sure to add to any article you review 🚀

kimkimani commented:
> @kimkimani great article - seems very informative - learned a lot about Nosql and SQL - wanted to ask if any of the other images needed sources (for example the joke image)
> 
> Please add those and then we can push your article to tech review 🚀
> 
> @cyberShaw - great work on the review - I added the peer review line at the bottom take a look and be sure to add to any article you review 🚀

Thank you @hectorkambow for the joke image, I code the idea online but restricted the wording and the background image.

cyberShaw commented:
> @kimkimani great article - seems very informative - learned a lot about Nosql and SQL - wanted to ask if any of the other images needed sources (for example the joke image)
> 
> Please add those and then we can push your article to tech review 🚀
> 
> @cyberShaw - great work on the review - I added the peer review line at the bottom take a look and be sure to add to any article you review 🚀

Sure thing! I'll be sure to add the peer review line to the articles I review.

kimkimani commented:
> > @kimkimani great article - seems very informative - learned a lot about Nosql and SQL - wanted to ask if any of the other images needed sources (for example the joke image)
> > Please add those and then we can push your article to tech review 🚀
> > @cyberShaw - great work on the review - I added the peer review line at the bottom take a look and be sure to add to any article you review 🚀
> 
> Thank you @hectorkambow for the joke image, I got the idea online but restricted the wording and the background image.

SQL database table was the only image I was in doubt, I removed some data but the rows and columns info remained the same

kimkimani commented:
I did the Relations image. Have added a source to the SQL database table. Should be fine now @hectorkambow

hectorkambow commented:
Good afternoon @sudosoph 👋 

Wanted to ask if you could give this "SQL or NoSQL " article a Final Tech review - feel free to include any tips you see fit to help educate our newer Peer Reviewers on the revision process and what they should be looking for moving forward. 🚀

Thank you

sudosoph reviewed (changes requested):
SQL Preliminary & Tech Review

* General:
    * Double check that images render.
    * May be helpful bolding new terms, like attributes, records, schemas
* What is SQL?: 
    * Introductory sentence is very generic.
    * Typo - “Data is the backbone of almost every modern busines”
    * Grammar - “The data in the tables can be related to each other hence the 'relational database’.”
        * Needs comma before ‘hence’ 
    * Consistent tone - “A collection of these multiple tables in a database is what we are calling a schema.”
        * Remove personal pronouns
    * Grammar & tone - “If we were to have a student info database, this is how we can represent a relation in an SQL database.”
        * Use full word “information”
        * Correction: “this is how we would…”
    * Define “foreign key mapping”
    * “Each row in the SQL database has its specific value characters, such as text and integers.”
        * Use proper article “a”
    * Elaborate: Why does SQL need a database management system (DBMS)?
* Table:
    * “SQL Relation output”
        * Capitalize caption
    * Courses: “Web Dev,React js”
        * Space missing
        * Capitalize JS
* What is NoSQL?: 
    * “NoSQL does not mean 'no SQL.’”
        * Remove
    * Define JSON.
    * Correct “student info” (see above)
* SQL vs NoSQL (High-Level Differences):
    * “Which is better?”
        * Remove.
    * Intro needs re-working. Rather than say “why choose this and not the other, and it’s okay” - don’t try to be reassuring to the reader. This is a technical article. "Let’s check some of the factors that will help you decide what fits your needs.” Is also unnecessary. Rather, provide an overview of the differences in your introduction to prepare the reader for your specific bullet points.
        * When to use SQL?:
            * “SQL will be a natural fit.”
                * Remove filler sentences like this.
            * “The biggest win for SQL is when the legacy…”
                * Start sentence at “The Legacy…”
            * “If your application requires multi-row transactions such as accounting and budget systems.”
                * Remove “if” = keep your bulleted list consistent by starting with parallel structure
            * “SQL will be a good match as it provides perfect…”
                * Condense: “SQL provides…”
            * Indent ACID section so it is clear it is a definition and not a reason to use SQL.
            * Use colons for what each letter in ACID represents.
            * Article should be “an”: “A SQL database system is designed”
            * “…to handle specific data types the way you want them to be”
                * Remove “the way you want them to be”
        * When to use NoSQL?:
            * Same feedback as previous section - begin bulleted list with same parallel structure
            * Same feedback as previous section - condense
                * “NoSQL will be a great fit as it offers” -> “NoSQL offers…”
                * Just the facts!
            * “NoSQL will help you check…”, “NoSQL will be a significant consideration”
                * Rather than using “will” repeatedly in the text, simply use the present tense verb of what it actually does rather than making promises to the reader.
* StackOverflow Database Insights:
    * Provide quick explanation for why Redis and MongoDB are most popular.
    * “Given MongoDB is still young, it is a database technology that most developers want to learn more about.”
        * Background? How old is MongoDB? Evidence for this?
    * “They both have their pros and cons. Taking advantage of their strengths would be the way to go.”
        * Filler. Remove.
    * “It would be best if you outlined your application model first and then decide which database model will fit your data structure.”
        * Tone - informal. Rather than giving advice directly to the reader, provide an objective statement on how to decide which database model fits the respective data structure.
    * “It should be a forward thinking decision because to migrate from SQL to NoSQL and vice verse can be a serious challenge, which is time-consuming and, at times costly.”
        * Rephrase & generalize to convey that migrating from one to the other is costly and time-consuming, and hence the differences should be considered while architecting software.

hectorkambow commented:
> SQL Preliminary & Tech Review
> 
> * General:
>   
>   * Double check that images render.
>   * May be helpful bolding new terms, like attributes, records, schemas
> * What is SQL?:
>   
>   * Introductory sentence is very generic.
>   * Typo - “Data is the backbone of almost every modern busines”
>   * Grammar - “The data in the tables can be related to each other hence the 'relational database’.”
>     
>     * Needs comma before ‘hence’
>   * Consistent tone - “A collection of these multiple tables in a database is what we are calling a schema.”
>     
>     * Remove personal pronouns
>   * Grammar & tone - “If we were to have a student info database, this is how we can represent a relation in an SQL database.”
>     
>     * Use full word “information”
>     * Correction: “this is how we would…”
>   * Define “foreign key mapping”
>   * “Each row in the SQL database has its specific value characters, such as text and integers.”
>     
>     * Use proper article “a”
>   * Elaborate: Why does SQL need a database management system (DBMS)?
> * Table:
>   
>   * “SQL Relation output”
>     
>     * Capitalize caption
>   * Courses: “Web Dev,React js”
>     
>     * Space missing
>     * Capitalize JS
> * What is NoSQL?:
>   
>   * “NoSQL does not mean 'no SQL.’”
>     
>     * Remove
>   * Define JSON.
>   * Correct “student info” (see above)
> * SQL vs NoSQL (High-Level Differences):
>   
>   * “Which is better?”
>     
>     * Remove.
>   * Intro needs re-working. Rather than say “why choose this and not the other, and it’s okay” - don’t try to be reassuring to the reader. This is a technical article. "Let’s check some of the factors that will help you decide what fits your needs.” Is also unnecessary. Rather, provide an overview of the differences in your introduction to prepare the reader for your specific bullet points.
>     
>     * When to use SQL?:
>       
>       * “SQL will be a natural fit.”
>         
>         * Remove filler sentences like this.
>       * “The biggest win for SQL is when the legacy…”
>         
>         * Start sentence at “The Legacy…”
>       * “If your application requires multi-row transactions such as accounting and budget systems.”
>         
>         * Remove “if” = keep your bulleted list consistent by starting with parallel structure
>       * “SQL will be a good match as it provides perfect…”
>         
>         * Condense: “SQL provides…”
>       * Indent ACID section so it is clear it is a definition and not a reason to use SQL.
>       * Use colons for what each letter in ACID represents.
>       * Article should be “an”: “A SQL database system is designed”
>       * “…to handle specific data types the way you want them to be”
>         
>         * Remove “the way you want them to be”
>     * When to use NoSQL?:
>       
>       * Same feedback as previous section - begin bulleted list with same parallel structure
>       * Same feedback as previous section - condense
>         
>         * “NoSQL will be a great fit as it offers” -> “NoSQL offers…”
>         * Just the facts!
>       * “NoSQL will help you check…”, “NoSQL will be a significant consideration”
>         
>         * Rather than using “will” repeatedly in the text, simply use the present tense verb of what it actually does rather than making promises to the reader.
> * StackOverflow Database Insights:
>   
>   * Provide quick explanation for why Redis and MongoDB are most popular.
>   * “Given MongoDB is still young, it is a database technology that most developers want to learn more about.”
>     
>     * Background? How old is MongoDB? Evidence for this?
>   * “They both have their pros and cons. Taking advantage of their strengths would be the way to go.”
>     
>     * Filler. Remove.
>   * “It would be best if you outlined your application model first and then decide which database model will fit your data structure.”
>     
>     * Tone - informal. Rather than giving advice directly to the reader, provide an objective statement on how to decide which database model fits the respective data structure.
>   * “It should be a forward thinking decision because to migrate from SQL to NoSQL and vice verse can be a serious challenge, which is time-consuming and, at times costly.”
>     
>     * Rephrase & generalize to convey that migrating from one to the other is costly and time-consuming, and hence the differences should be considered while architecting software.

@cyberShaw  Please also review the comments made by one of our veteran Peer Reviewers - to get a better understanding on how we can draw more value and help students correct their articles moving forward! 🚀

cyberShaw commented:
Thank you for the feedback, @sudosoph ! I'll make sure to implement these into my peer reviews.

cyberShaw commented:
@kimkimani Please 

> SQL Preliminary & Tech Review
> 
> * General:
>   
>   * Double check that images render.
>   * May be helpful bolding new terms, like attributes, records, schemas
> * What is SQL?:
>   
>   * Introductory sentence is very generic.
>   * Typo - “Data is the backbone of almost every modern busines”
>   * Grammar - “The data in the tables can be related to each other hence the 'relational database’.”
>     
>     * Needs comma before ‘hence’
>   * Consistent tone - “A collection of these multiple tables in a database is what we are calling a schema.”
>     
>     * Remove personal pronouns
>   * Grammar & tone - “If we were to have a student info database, this is how we can represent a relation in an SQL database.”
>     
>     * Use full word “information”
>     * Correction: “this is how we would…”
>   * Define “foreign key mapping”
>   * “Each row in the SQL database has its specific value characters, such as text and integers.”
>     
>     * Use proper article “a”
>   * Elaborate: Why does SQL need a database management system (DBMS)?
> * Table:
>   
>   * “SQL Relation output”
>     
>     * Capitalize caption
>   * Courses: “Web Dev,React js”
>     
>     * Space missing
>     * Capitalize JS
> * What is NoSQL?:
>   
>   * “NoSQL does not mean 'no SQL.’”
>     
>     * Remove
>   * Define JSON.
>   * Correct “student info” (see above)
> * SQL vs NoSQL (High-Level Differences):
>   
>   * “Which is better?”
>     
>     * Remove.
>   * Intro needs re-working. Rather than say “why choose this and not the other, and it’s okay” - don’t try to be reassuring to the reader. This is a technical article. "Let’s check some of the factors that will help you decide what fits your needs.” Is also unnecessary. Rather, provide an overview of the differences in your introduction to prepare the reader for your specific bullet points.
>     
>     * When to use SQL?:
>       
>       * “SQL will be a natural fit.”
>         
>         * Remove filler sentences like this.
>       * “The biggest win for SQL is when the legacy…”
>         
>         * Start sentence at “The Legacy…”
>       * “If your application requires multi-row transactions such as accounting and budget systems.”
>         
>         * Remove “if” = keep your bulleted list consistent by starting with parallel structure
>       * “SQL will be a good match as it provides perfect…”
>         
>         * Condense: “SQL provides…”
>       * Indent ACID section so it is clear it is a definition and not a reason to use SQL.
>       * Use colons for what each letter in ACID represents.
>       * Article should be “an”: “A SQL database system is designed”
>       * “…to handle specific data types the way you want them to be”
>         
>         * Remove “the way you want them to be”
>     * When to use NoSQL?:
>       
>       * Same feedback as previous section - begin bulleted list with same parallel structure
>       * Same feedback as previous section - condense
>         
>         * “NoSQL will be a great fit as it offers” -> “NoSQL offers…”
>         * Just the facts!
>       * “NoSQL will help you check…”, “NoSQL will be a significant consideration”
>         
>         * Rather than using “will” repeatedly in the text, simply use the present tense verb of what it actually does rather than making promises to the reader.
> * StackOverflow Database Insights:
>   
>   * Provide quick explanation for why Redis and MongoDB are most popular.
>   * “Given MongoDB is still young, it is a database technology that most developers want to learn more about.”
>     
>     * Background? How old is MongoDB? Evidence for this?
>   * “They both have their pros and cons. Taking advantage of their strengths would be the way to go.”
>     
>     * Filler. Remove.
>   * “It would be best if you outlined your application model first and then decide which database model will fit your data structure.”
>     
>     * Tone - informal. Rather than giving advice directly to the reader, provide an objective statement on how to decide which database model fits the respective data structure.
>   * “It should be a forward thinking decision because to migrate from SQL to NoSQL and vice verse can be a serious challenge, which is time-consuming and, at times costly.”
>     
>     * Rephrase & generalize to convey that migrating from one to the other is costly and time-consuming, and hence the differences should be considered while architecting software.

@kimkimani Please do incorporate these changes so that we can move forward

kimkimani commented:
@sudosoph @cyberShaw got some updates.
